src/spec/payid-easy-checkout.md

```diff
@@ -0,0 +1,330 @@
+
+## PayID Easy Checkout Wallet Discovery
+The merchant site would resolve the customer's PayID to a host as defined in [PAYID-URI][], in this case resolving
+alice$wallet.com to https://wallet.com. The merchant site would then perform PayID Discovery as defined in
+[PAYID-DISCOVERY][] to receive a PayID Easy Checkout JRD like this:
+    
+    GET /.well-known/webfinger?resource=payid%3Aalice%24wallet.com
+    Host: wallet.com
+    
+If the server has enabled PayID Easy Checkout in their wallet, they would respond with something like this:
+     
+     HTTP/1.1 200 OK
+     Access-Control-Allow-Origin: *
+     Content-Type: application/jrd+json
+
+     {
+       "subject" : "payid:alice$wallet.com",
+       "links" :
+       [
+         {  
+           "rel": "https://payid.org/ns/payid-easy-checkout/1.0",
+           "template": "https://wallet.com/checkout?amount={amount}&receiverPayId={receiverPayId}&currency={currency}&nextUrl={nextUrl}"
+         }
+       ]
+     }
+
+## Expand Wallet Discovery URL Template
+The merchant would parse the PayID Discovery response and iterate over the "links" collection to find the link with 
+the Relation Type of "https://payid.org/ns/payid-easy-checkout/1.0". The site can then do a search and replace on
+the "template" field value in the link, replacing all occurrences of the predefined query parameter template names with 
+the values they want to send to the customer's wallet. One query parameter of note is the "nextUrl" parameter, which
+allows the merchant to supply a redirect or callback URL for the sender's wallet to call once the customer has confirmed
+the payment.
+
+## Redirect Customer to Their Wallet
+Once the merchant populates the required query parameters in the URL template, they would redirect the customer to 
+the resulting URL. In this example, the merchant would like to display a "Thank You" page, and replaces `{nextUrl}` with 
+`https://wallet.com/checkout?amount=10&receiverPayId=payid%2Apay%24merchant.com&currency=XRP&nextUrl=https://merchant.com/thankyou`.
+
+## Customer Confirms Payment
+After the customer clicks the "Pay with PayID" button the merchant's site, and the merchant performs the previous steps,
+the customer will be redirected to their wallet at the URL from the previous step.  The wallet front end can
+read the query parameters from the redirect URL and render a confirmation page or modal with all of the required fields
+pre-populated.
+
+Once the customer confirms the payment, the wallet would perform a PayID address lookup on the "receiverPayId" query
+parameter to get the payment address of the merchant and submit a transaction to the underlying ledger or payment system.
+The merchant can then redirect the user back to the URL specified in the "nextUrl" query parameter, which will display
+the "Thank You" page of the merchant.
+
+# PayID Easy Checkout Protocol
+The PayID Easy Checkout Protocol can be used to facilitate an end-to-end checkout flow for users transacting
```

Comment:
I think we should try to limit the protocol to the machines/software that need to operate the protocol. E.g., maybe instead of "user" we say "client" (which can be a browser, phone, etc). Not sure what to call the "merchant" -- maybe the "recipient"?

Reply:
Would both the merchant and wallet have a "client"? So something like:
`The PayID Easy Checkout Protocol can be used to facilitate an end-to-end checkout flow between a recipient client and a wallet client?`